concepts/05 UI Components/Gantt/20 Filter Data/2 Header Filter.md

```diff
@@ -0,0 +1,472 @@
+                    caption="Subject"
+                    filterType="exclude">
+                    
+                    <HeaderFilter
+                        searchMode="startswith">
+                    </HeaderFilter>
+
+                </Column>
+                
+                <Column
+                    dataField="start"
+                    caption="Start Date"
+                    allowHeaderFiltering={false}>
+                </Column>
+
+                <Column 
+                    dataField="end"
+                    caption="End Date">
+
+      	            <HeaderFilter
+                        groupInterval="month"
+                    />
+
+                </Column>
+                {/* ... */}
+            </Gantt>
+        );
+    };
+
+   export default App;
+
+##### ASP.NET Core Controls
+
+    <!--Razor C#-->
+    @(Html.DevExtreme().Gantt()
+        .Columns(columns => {
+            columns.AddFor(m => m.Title)
+                .DataField("title")
+                .Caption("Subject")
+                .FilterType(FilterType.Exclude)
+                .HeaderFilter(headerFilter => headerFilter.SearchMode(CollectionSearchMode.StartsWith));
+
+            columns.AddFor(m => m.Title)
+                .DataField("start")
+                .Caption("Start Date")
+                .AllowHeaderFiltering(false)
+            
+            columns.AddFor(m => m.Title)
+                .DataField("end")
+                .Caption("End Date")
+                .HeaderFilter(headerFilter => headerFilter.GroupInterval(HeaderFilterGroupInterval.Month));
+            @* ... *@
+        })
+        .HeaderFilter(e => {
+            e.Visible(true)
+            e.allowSearch(true);
+        })
+        
+        // ...
+    )
+
+##### ASP.NET MVC Controls
+
+    <!--Razor C#-->
+    @(Html.DevExtreme().Gantt()
+        .Columns(columns => {
+            columns.AddFor(m => m.Title)
+                .DataField("title")
+                .Caption("Subject")
+                .FilterType(FilterType.Exclude)
+                .HeaderFilter(headerFilter => headerFilter.SearchMode(CollectionSearchMode.StartsWith));
+
+            columns.AddFor(m => m.Title)
+                .DataField("start")
+                .Caption("Start Date")
+                .AllowHeaderFiltering(false)
+            
+            columns.AddFor(m => m.Title)
+                .DataField("end")
+                .Caption("End Date")
+                .HeaderFilter(headerFilter => headerFilter.GroupInterval(HeaderFilterGroupInterval.Month));
+            @* ... *@
+        })
+        .HeaderFilter(e => {
+            e.Visible(true);
+            e.allowSearch(true);
+        })
+        
+        // ...
+    )
+
+---
+
+The **Gantt** allows you to define initial filter settings in code. Specify a column's [filterValues](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterValues) property to apply the filter criteria to the column. 
```

Comment:
> The **Gantt** allows you to define initial filter settings in code. Specify a column's [filterValues](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterValues) property to apply **the** filter criteria to the column.

Of which filter criteria do we speak exactly?

Reply:
I've taken the definite article based on the previously mentioned initial filter settings. Such as "we can specify some initial filter settings, and to specify the filter criteria (concrete/defined), use..". But I'm not sure about that logic, so I would like to listen your opinion

Reply:
That depends on what did you want to say here. If you do not need an emphasis on specific settings (initial, etc.), then you can just omit the definite article at "filter criteria" and be done with it. If you need to emphasize the fact that you customize the initial filter settings, just say so. For example:

```suggestion
The **Gantt** allows you to define initial filter settings in code. Specify a column's [filterValues](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterValues) property to customize the column's initial filter settings. 
```

Just remember that there's no connection between "settings" and "criteria" until you explicitly declare it.